Automated cherry pick of #41455

gnufied opened the pull request:
Cherry pick of #41455 on release-1.4.

#41455: Fix AWS device allocator to only use valid device names

gnufied committed:
Fix AWS device allocator to only use valid device names

According to
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
we can only use /dev/xvd[b-c][a-z] as device names - so we can only
allocate upto 52 ebs volumes on a node.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/41492)
<!-- Reviewable:end -->

gnufied commented:
cc @saad-ali @justinsb

gnufied commented:
@jingxu97 need your help getting this merged as well.

jessfraz commented:
@k8s-bot unit test this

jessfraz commented:
@k8s-bot cvm gce e2e test this

gnufied commented:
@jessfraz I have rebased this branch and that cancelled /lgtm and /approve. I rebased with release-1.4 hoping that would fix the build failures I was seeing on this branch. PTAL.
